sklearn/model_selection/_split.py

```diff
@@ -758,8 +758,8 @@ class StratifiedGroupKFold(_BaseKFold):
     return stratified folds with non-overlapping groups. The folds are made by
     preserving the percentage of samples for each class.
 
-    The same group will not appear in two different folds (the number of
-    distinct groups has to be at least equal to the number of folds).
+    Each group will appear exactly once in the test set across all folds (the
+    number of distinct groups has to be at least equal to the number of folds).
 
     The difference between GroupKFold and StratifiedGroupKFold is that
```

Comment:
I think cross-referencing classes in doc-strings makes browsing the documentation easier. (I can't suggest for the rest of the paragraph.)  
```suggestion
    The difference between :class:`~sklearn.model_selection.GroupKFold`
    and :class:`~sklearn.model_selection.StratifiedGroupKFold` is that
```

Note that this phrasing is also present in other `_BaseKFold`'s subclasses' doc-strings, and could be adapted similarly in another PR.

Reply:
Hi @jjerphan, sure, make sense! Thanks for the suggestion, just committed it